File: backend/models/application_model.js
const mongoose = require('mongoose');
const db = require('../configuration/db');

const {Schema} = mongoose;

const applicationSchema = new Schema({
    companyId :{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'company',
        required: true,
    },

    companyName : {
        type: String,
        required: true,
    },

    studentSapid : {
        type: Number, 
        required: true,
    },

    nameStudent : {
        type: String,
        required: true,
    },

    middlnameStudent : {
        type: String,
        required : true,
    },

    surnameStudent : {
        type: String,
        required : true,
    },

    studentBranch:{
        type: String,
        required : true,
    },

    studentGpa : {
        type: Number,
        required  : true,
    },

    student10th : {
        type: Number,
        required  : true,
    },

    student12th : {
        type: Number,
        required  : true,
    },

    studentResume : {
        data: Buffer, // Store binary data
        contentType: String,
    }
    
});

const  applicationModel = db.model('application', applicationSchema);

module.exports = applicationModel;

File: frontend/lib/models/application_model.dart
class Application {
  final String companyId;
  final String companyName;
  final int studentSapid;
  final String nameStudent;
  final String middlenameStudent;
  final String surnameStudent;
  final String studentBranch;
  final double studentGpa;
  final int student10th;
  final int student12th;
  final String studentResume;

  Application({
    required this.companyId,
    required this.companyName,
    required this.studentSapid,
    required this.nameStudent,
    required this.middlenameStudent,
    required this.surnameStudent,
    required this.studentBranch,
    required this.studentGpa,
    required this.student10th,
    required this.student12th,
    required this.studentResume,
  });

  factory Application.fromJson(Map<String, dynamic> json) {
    return Application(
      companyId: json['companyId'],
      companyName: json['companyName'],
      studentSapid: json['studentSapid'],
      nameStudent: json['nameStudent'],
      middlenameStudent: json['middlenameStudent'],
      surnameStudent: json['surnameStudent'],
      studentBranch: json['studentBranch'],
      studentGpa: json['studentGpa'].toDouble(),
      student10th: json['student10th'],
      student12th: json['student12th'],
      studentResume: json['studentResume'],
    );
  }
}
